# 08 · CFTR2 — the disease-specific *functional* truth set

[CFTR2](https://cftr2.org) classifies *CFTR* variants using **patient outcomes + in-vitro CFTR function assays** — a different, more *functional* kind of evidence than ClinVar's clinical assertions. That functional axis makes it an **orthogonal** truth set (notebook 13). This toolkit ships the **REAL** full CFTR2 release.

> ✅ **REAL data** — the full public CFTR2 list (30 January 2026 release, ~2,097 variants), shipped at `data/cftr2_2026-01-30.csv`. Redistributed under cftr2.org's public data-use terms; please cite CFTR2 if you use it.

In [1]:
import sys, pathlib
sys.path.insert(0, str(pathlib.Path.cwd().parent))
import toolkit as tk
import pandas as pd, numpy as np
%matplotlib inline

## 5 · CFTR2 — a disease-specific, *functional* reference

[CFTR2](https://cftr2.org) is different in kind from ClinVar. It is a **CF-specific** database that classifies *CFTR* variants as:

- **CF-causing**
- **Varying clinical consequence** (formerly "CF-causing (mild)")
- **Non CF-causing**
- **No interpretation available** — not yet enough evidence

Crucially, CFTR2's calls are built from **two kinds of evidence together**:
1. **Patient data** — real clinical outcomes across thousands of people with CF who carry the variant.
2. **In-vitro CFTR function assays** — measuring, in the lab, how much working chloride-channel the variant protein actually produces.

That second, functional axis is what makes CFTR2 an *orthogonal* truth set — see notebook 13.

In [2]:
cftr2 = tk.load_cftr2()          # REAL — full CFTR2 30 January 2026 release
print('source :', cftr2['source'].unique(), '| variants:', len(cftr2),
      '| with a missense key:', (cftr2['protein_variant'] != '').sum())
print()
print(cftr2['cftr2_class'].value_counts().to_string())
cftr2.head(8)

source : ['REAL'] | variants: 2097 | with a missense key: 780

cftr2_class
CF-causing                      1245
No interpretation available      722
Varying clinical consequence      83
Non CF-causing                    42


,protein_variant,legacy_name,protein_name,cdna_name,cftr2_alleles,cftr2_af,cftr2_class,grch38_chr,grch38_pos,grch38_ref,grch38_alt,source
0,,F508del,p.Phe508del,c.1521_1523del,137363,0.650682595473364,CF-causing,7.0,117559590.0,ATCT,A,REAL
1,,G542X,p.Gly542X,c.1624G>T,5752,0.027246975453089916,CF-causing,7.0,117587778.0,G,T,REAL
2,G551D,G551D,p.Gly551Asp,c.1652G>A,3831,0.01814728146049852,CF-causing,7.0,117587806.0,G,A,REAL
3,N1303K,N1303K,p.Asn1303Lys,c.3909C>G,3551,0.01682093355944407,CF-causing,7.0,117652877.0,C,G,REAL
4,,W1282X,p.Trp1282X,c.3845G>A|c.3846G>A,2500,0.011842391973700416,CF-causing,NaN,NaN,NaN,NaN,REAL
5,R117H,R117H,p.Arg117His,c.350G>A,2262,0.010714996257804137,Varying clinical consequence,7.0,117530975.0,G,A,REAL
6,,3849+10kbC->T,p.?,c.3718-2477C>T,1990,0.00942654401106553,CF-causing,7.0,117639961.0,C,T,REAL
7,,621+1G->T,p.?,c.489+1G>T,1860,0.008810739628433109,CF-causing,7.0,117531115.0,G,T,REAL


> ✅ **This is the REAL CFTR2 list** — the full public 30 January 2026 release (~2,097 variants), shipped at `data/cftr2_2026-01-30.csv` and built from the cftr2.org download by `build_cftr2.py`. A 1-letter `protein_variant` key (e.g. `G551D`) is derived for the ~780 simple-missense variants so CFTR2 class joins straight onto the AlphaMissense / gnomAD tables; non-missense rows keep their legacy / cDNA names and GRCh38 coordinates. CFTR2 data is redistributed under cftr2.org's public data-use terms — please cite CFTR2 if you use it.

## 6 · ★ Key teaching point: why CFTR2 is a *more orthogonal* truth set

Here is the subtle trap that this whole notebook is building toward.

Many predictors — especially **supervised** ones like **REVEL** — were *trained* using variant labels that ultimately trace back to clinical databases in the **ClinVar lineage**. So if you 'benchmark' REVEL against ClinVar, you may partly be testing it **on data related to what it learned from**. It can look great for a circular reason: it's grading its own homework. This is called **circularity** (or label leakage).

**CFTR2 breaks that loop.** Because its calls incorporate **functional assay** evidence — a wet-lab measurement of channel function that a sequence model *never saw during training* — CFTR2 is a more **orthogonal** (independent) yardstick for judging sequence predictors. A predictor that agrees with CFTR2 is agreeing with *biology measured in a dish*, not merely with prior clinical opinion.

> **Rule of thumb:** use ClinVar for **breadth**, but lean on CFTR2 (and other functional data) when you need an **independent** check — particularly for supervised tools. We put this into practice in **notebook 13**, where we benchmark the predictors against both truth sets and watch which tools hold up when the yardstick is orthogonal.

## Key takeaways

1. **CFTR2** calls combine **patient data + functional assays** → a more *orthogonal* truth set than ClinVar for benchmarking sequence predictors.
2. Now shipped **REAL** (~2,097 variants; 780 with a 1-letter missense key, 779 of which join to real AlphaMissense).
3. The **CFTR2-vs-ClinVar agreement** cross-check lives in the integration notebook (12).

**Next:** notebook 09 — **SpliceAI**, the first splice predictor.